client/src/www/model/platform_error.ts

```diff
@@ -158,7 +200,7 @@ export class PlatformError extends CustomError {
    *   throw PlatformError.parseFrom(e);
    * }
    */
-  static parseFrom(errObj: string | Error | unknown): PlatformError {
+  static parseFrom(errObj: string | Error | unknown): Error {
```

Comment:
Maybe `parseToError` reads better?

Reply:
Renamed to `deserializeError` and moved it out of the class for readability.

In the process, I realized that we were calling it twice in the Cordova MethodChannel. So I'm now calling it in the pluginExec instead, to ensure we always deserialize the error properly when we call native code.

I also deleted the pluginExecWithCode, which was doing the same thing. It's legacy from the usage of error codes.